app/services/AuditService.scala

```diff
@@ -190,6 +177,26 @@ class AuditService @Inject() (auditConnector: AuditConnector)(implicit ec: Execu
     )
   }
 
+  private def toPaymentPlanSetupAuditDetail(
+      journey:         AgreedTermsAndConditions,
+      responseFromTtp: Either[HttpException, ArrangementResponse]
+  )(implicit r: AuthenticatedJourneyRequest[_]): PaymentPlanSetUpAuditDetail = {
+    val maybeArrangementResponse: Option[ArrangementResponse] = responseFromTtp.toOption
+    val status: Int = responseFromTtp.fold(something => something.responseCode, _ => Status.ACCEPTED)
```

Comment:
there's probably a better name than `something` here 😆 you probably could just do `_.responseCode` though and not need a name

Reply:
haha woops 😂